File: 2024/10 - October/05. Not a subset sum/Solution.java
class Solution 
{
    public long findSmallest(int[] arr) 
    {
        long smallest_missing = 1;
        
        for(int i=0; i<arr.length; i++)
        {
            if(arr[i] > smallest_missing) break;
            
            smallest_missing += arr[i];
        }
        
        return smallest_missing;
    }
}

File: 2024/10 - October/05. Not a subset sum/Solution.cpp
class Solution 
{
  public:
    long long findSmallest(vector<int> &arr) 
    {
        long long smallest_missing = 1;
        
        for(int i=0; i<arr.size(); i++)
        {
            if(arr[i] > smallest_missing)
            {
                break;
            }
            
            smallest_missing += arr[i];
        }
        
        return smallest_missing;
    }
};
